### DISCUSSION TITLE
Read only access for PNP app registration

### DISCUSSION THREAD

**Discussion by fusrohdan** [pnp/powershell#4560] [Q&A]
Hey,

So I have created an app registration which I can connect to successfully.  Command used to connect is:
Connect-PnPOnline TENANT-admin.sharepoint.com -ClientId CLIENTID -Tenant TENANT.onmicrosoft.com -CertificatePath C:\it\PowershellPnP.pfx -CertificatePassword $password 

This connects fine, however when I run the following command I get a 401 unauthorised error
Get-PnPTenantSite
Get-PnPTenantSite: The remote server returned an error: (401) Unauthorized.

Any other pnp command I use returns the same.  
I have granted the app registration the following permissions
![image](https://github.com/user-attachments/assets/3d72ab2b-beab-4426-8d61-4067d0580a49)
![image](https://github.com/user-attachments/assets/3856f52b-6ec5-4442-bc15-aeeae3482d31)

What I am currently trying to achieve is to grab a list of all sites, storage used, and get the content of the description of the site. 

How can I achieve this using pnp?

Thanks!

**Comment by enricobeyer** [comment]
Hello fusrohdan,

first try the command **_Register-PnPEntraIDAppForInteractiveLogin_** to Login with -Interactive.
Then connect with
**_Connect-PnPOnline -ClientId $ClientID -Url tenant-admin.sharepoint.com/ -Interactive_**
First you may try it without **_-Tenant TENANT.onmicrosoft.com_**

Best regards,
Enrico

**Comment by Barbarur** [comment]
PnPTenantSite uses SPO API.
You are connecting using app credentials. But the SPO API is delegated.

Either you connect using delegated credentials or you grant application permissions to SPO API.

Keep in mind that application SPO API can only be used with a certification. Secret ID wouldn't allow you to connect.

**Comment by prawnz** [comment]
Since you are calling the tenant cmdlet commands , I assumed you have been given a SharePoint Administrator or a Global Administrator role?
